src/libostree/ostree-bootconfig-parser.c

```diff
@@ -56,6 +59,96 @@ ostree_bootconfig_parser_clone (OstreeBootconfigParser *self)
   return parser;
 }
 
+/**
+ * ostree_bootconfig_parser_parse_tries:
+ * @self: Parser
+ * @path: File path
+ *
+ * Parses a suffix of two counters in the form "+LEFT-DONE" from the end of the
+ * filename (excluding file extension).
+ */
+static void
+parse_bootloader_tries (const char *filename, guint64 *out_left, guint64 *out_done, GError **error)
+{
+  gchar *counter = NULL;
+  gchar *old_counter = NULL;
+  *out_left = 0;
+  *out_done = 0;
+
+  for (;;)
+    {
+      char *plus = strchr (counter ?: filename, '+');
+      if (plus)
+        {
+          /* Look for the last "+" symbol in the filename */
+          counter = plus + 1;
+          continue;
+        }
+      if (counter)
+        break;
+
+      /* No boot counter found */
+      return;
+    }
+
+  guint64 tries_left, tries_done = 0;
+
+  old_counter = counter;
+  tries_left = g_ascii_strtoull (old_counter, &counter, 10);
```

Comment:
I think this is also OK but it would seem clearer to me to leave the value of `counter` pointing where it is and then assign to a new `tries` variable? Like:

```
const char *tries = NULL;
tries_left = g_ascii_strtoull (counter, &tries, 10);
...
```
or so?

Reply:
I quickly looked into that and created a prototype, but it seems to only make the code more complex IMO